### DISCUSSION TITLE
VCPKG tested with MacOS

### DISCUSSION THREAD

**Discussion by remoe** [pelicanmapping/rocky#9] [Q&A / Support]
I've tried to build this project with VCPKG. It failed.

Steps:

1)

added `set(VCPKG_MANIFEST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/full)` in CMakeLists.txt

2)

Need to install:
- pkg-config
- autoconf
- automake 
- libtool 

with brew or other package manager. Don't know why this needed with VCPKG :)
Check if all this tools in PATH environment variable.

3)

call cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=${vcpkgdir}/scripts/buildsystems/vcpkg.cmake 

4)

The following packages will be built and installed:

```
  * assimp[core]:arm64-osx -> 5.2.5
  * boost-assert[core]:arm64-osx -> 1.82.0#1
  * boost-config[core]:arm64-osx -> 1.82.0#1
  * boost-core[core]:arm64-osx -> 1.82.0#1
  * boost-move[core]:arm64-osx -> 1.82.0#1
  * boost-smart-ptr[core]:arm64-osx -> 1.82.0#1
  * boost-static-assert[core]:arm64-osx -> 1.82.0#1
  * boost-throw-exception[core]:arm64-osx -> 1.82.0#1
  * boost-type-traits[core]:arm64-osx -> 1.82.0#1
  * boost-uninstall[core]:arm64-osx -> 1.82.0#1
  * boost-vcpkg-helpers[core]:arm64-osx -> 1.82.0#1
  * draco[core]:arm64-osx -> 1.5.6
  * expat[core]:arm64-osx -> 2.5.0#3
  * freetype[brotli,bzip2,core,png,zlib]:arm64-osx -> 2.12.1#3
    gdal[core,curl,default-features,expat,geos,gif,hdf5,iconv,jpeg,lerc,libkml,libspatialite,libxml2,lzma,netcdf,openjpeg,openssl,pcre2,png,postgresql,qhull,recommended-features,sqlite3,webp,zstd]:arm64-osx -> 3.7.0#2
  * giflib[core]:arm64-osx -> 5.2.1#3
    glm[core]:arm64-osx -> 0.9.9.8#2
  * glslang[core]:arm64-osx -> 12.2.0
  * json-c[core]:arm64-osx -> 2022-06-26#3
  * kubazip[core]:arm64-osx -> 0.2.4
  * lerc[core]:arm64-osx -> 4.0#1
  * libgeotiff[core]:arm64-osx -> 1.7.1#3
  * libkml[core]:arm64-osx -> 1.3.0#12
  * libspatialite[core,freexl,geocallbacks]:arm64-osx -> 5.0.1#12
  * minizip[core]:arm64-osx -> 1.2.13#2
  * poly2tri[core]:arm64-osx -> 2020-07-21#3
  * pugixml[core]:arm64-osx -> 1.13.0
  * rapidjson[core]:arm64-osx -> 2023-04-27
  * stb[core]:arm64-osx -> 2023-04-11#1
    tinyxml[core]:arm64-osx -> 2.6.2#10
  * uriparser[core]:arm64-osx -> 0.9.7
  * utfcpp[core]:arm64-osx -> 3.2.3
    vsg[core]:arm64-osx -> 1.0.5 -- /Users/x/src/rocky/vcpkg/ports/vsg
    vsgimgui[core]:arm64-osx -> 0.0.0 -- /Users/x/src/rocky/vcpkg/ports/vsgimgui
    vsgxchange[core]:arm64-osx -> 1.0.1 -- /Users/x/src/rocky/vcpkg/ports/vsgxchange
  * vulkan[core]:arm64-osx -> 1.1.82.1#6
  * 
```

5)

It fails here:

libspatialite

The issue is reported.

**Comment by plevy** [comment]
Yeah, the initial is windows centric and uses a VCPKG_DIR env variable.  

If you go to your vcpkg folder and 
vcpkg install libspatialite:arm64-osx

Does it also fail?  In theory, packages are tested on multiple platforms

Thank you for the help

**Comment by remoe** [comment]
`vcpkg install libspatialite:arm64-osx --recurse` failed as well.

@plevy why is `bootstrap-vcpkg.bat` needed?

My suggestion: 
- add `set(VCPKG_MANIFEST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/full)` to CMakeLists.txt
- document the other things in readme ?

Because the build process should work on multiple platforms like what one found in VSG?
